carbon/config.py

```diff
@@ -1,73 +1,96 @@
 import json
 import logging
 import os
+from collections.abc import Iterable
 
 import sentry_sdk
 
-ENV_VARS = [
-    "FEED_TYPE",
-    "SNS_TOPIC",
-    "SYMPLECTIC_FTP_PATH",
-    "WORKSPACE",
-    "DATAWAREHOUSE_CLOUDCONNECTOR_JSON",
-    "SYMPLECTIC_FTP_JSON",
-]
+root_logger = logging.getLogger()
 
 
-def configure_logger(logger: logging.Logger, log_level_string: str) -> str:
-    if log_level_string.upper() not in logging.getLevelNamesMapping():
-        invalid_logging_level_message = (
-            f"'{log_level_string}' is not a valid Python logging level"
-        )
-        raise ValueError(invalid_logging_level_message)
-    log_level = logging.getLevelName(log_level_string.upper())
-    if log_level < 20:  # noqa: PLR2004
-        logging.basicConfig(
-            format="%(asctime)s %(levelname)s %(name)s.%(funcName)s() line %(lineno)d: "
-            "%(message)s"
-        )
-        logger.setLevel(log_level)
-        for handler in logging.root.handlers:
-            handler.addFilter(logging.Filter("carbon"))
-    else:
-        logging.basicConfig(
-            format="%(asctime)s %(levelname)s %(name)s.%(funcName)s(): %(message)s"
-        )
-        logger.setLevel(log_level)
-    return (
-        f"Logger '{logger.name}' configured with level="
-        f"{logging.getLevelName(logger.getEffectiveLevel())}"
+class Config:
+    REQUIRED_ENVIRONMENT_VARIABLES: Iterable[str] = (
+        "FEED_TYPE",
+        "DATAWAREHOUSE_CLOUDCONNECTOR_JSON",
+        "SYMPLECTIC_FTP_JSON",
+        "SYMPLECTIC_FTP_PATH",
+        "SNS_TOPIC_ARN",
+        "WORKSPACE",
     )
+    FEED_TYPE: str
+    CONNECTION_STRING: str
+    SYMPLECTIC_FTP_USER: str
+    SYMPLECTIC_FTP_PASS: str
+    SYMPLECTIC_FTP_HOST: str
+    SYMPLECTIC_FTP_PORT: str
+    SYMPLECTIC_FTP_PATH: str
+    SNS_TOPIC_ARN: str
+    WORKSPACE: str
 
+    def __init__(
+        self,
+        log_level: str = "INFO",
+    ) -> None:
+        self.log_level = log_level
+        self.configure_logger()
+        self.load_environment_variables()
+        self.configure_sentry()
```

Comment:
_Really_ minor here, but as we're talking about the `Config` class...

I might propose a blank line between the attributes passed and set, and these methods that perform initialization logic.  Just a single blank line might be enough to set them apart, e.g.

```python
self.log_level = log_level

self.configure_logger()
self.load_environment_variables()
self.configure_sentry()
```

Reply:
Also agree about this!

Reply:
Blank line has been added!